### PR TITLE
feat: exception logger

### PR DESCRIPTION
**Pull request by thorseraq**

### CODE REVIEW

File: apps/server/src/middleware/exception-logger.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@Catch(HttpException)
+export class ExceptionLogger implements ExceptionFilter {
+  private logger = new Logger('ExceptionLogger');
+
+  catch(exception: Error, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const request = ctx.getRequest<Request>();
+    const requestId = request?.header(REQUEST_ID);
+    this.logger.error(
+      `${requestId ? `requestId-${requestId}:` : ''}${exception.message}`,
```
Comment (Brooooooklyn) [inline apps/server/src/middleware/exception-logger.ts:21]:
```suggestion
      new Error(`${requestId ? `requestId-${requestId}:` : ''}${exception.message}`, { cause: exception })
```

File: apps/server/src/middleware/exception-logger.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      exception.stack
```
Comment (Brooooooklyn) [inline apps/server/src/middleware/exception-logger.ts:22]: remove this line

File: apps/server/src/modules/index.ts [hunk 3]
```diff
@@ -37,4 +40,14 @@ switch (SERVER_FLAVOR) {
     break;
 }
 
-export { BusinessModules };
+const Providers: Provider[] = [];
+
+if (NODE_ENV === 'production') {
+  console.log('push app filter');
```
Comment (Brooooooklyn) [inline apps/server/src/modules/index.ts:46]: remove

File: apps/server/src/modules/index.ts [hunk 4]
```diff
@@ -37,4 +40,13 @@ switch (SERVER_FLAVOR) {
     break;
 }
 
-export { BusinessModules };
+const Providers: Provider[] = [];
+
+if (NODE_ENV === 'production') {
```
Comment (forehalo) [inline apps/server/src/modules/index.ts:45]: why production only?

File: apps/server/src/middleware/exception-logger.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+
+import { REQUEST_ID } from '../constants';
+
+@Catch(HttpException)
```
Comment (forehalo) [inline apps/server/src/middleware/exception-logger.ts:12]: what about other errors

Comment (thorseraq) [reply apps/server/src/middleware/exception-logger.ts:12]: we don't have rpc request, for ws request I suggest don't catch them, make socketIO handle it

File: apps/server/src/modules/index.ts (hunk 4 above)
Comment (thorseraq) [reply apps/server/src/modules/index.ts:45]: It makes test output ugly

Comment (thorseraq) [reply apps/server/src/modules/index.ts:45]: how about NODE_ENV !== 'test' https://github.com/toeverything/AFFiNE/blob/cb0d9fa22aaf5eb130ebc8ba92b72e23b9a76bcd/apps/server/scripts/run-test.ts#L25

**Review by Brooooooklyn — APPROVED** [review]

**Pull request merged**